cmake/private/FairRootTargets.cmake

```diff
@@ -52,3 +53,33 @@ function(fairroot_install_exported)
     ARCHIVE DESTINATION ${PROJECT_INSTALL_LIBDIR}
     LIBRARY DESTINATION ${PROJECT_INSTALL_LIBDIR})
 endfunction()
+
+
+function(fairroot_discover_catch2_tests target)
+  cmake_parse_arguments(PARSE_ARGV 1 ARGS "" "SUITE" "")
+
+  set(test_prefix "")
+  if(ARGS_SUITE)
+    set(test_prefix "${ARGS_SUITE}::")
+  endif()
+
+  # see https://github.com/FairRootGroup/FairRoot/issues/1523
+  set(dl_paths "")
+  set(req_version 3.22)
+  if(    ${CMAKE_SYSTEM_NAME} STREQUAL "Linux"
+     AND target STREQUAL test_ExSimulation1)
+    set(dl_paths ${LD_LIBRARY_PATH})
+  endif()
```

Comment:
Opinions?

Reply:
I don't like the dependency on test names in "generic" code. We learned how that fails for "labels" in our CI lately.

I would prefer something more explicit.

In other words: In six months, everybody will have forgotten about this code and if we rename `ExSimulation1` at that point, it will end up in strange things.

Reply:
Sure, however the CI-label example was yet different: There we somewhat broke user-facing abstractions. Meaning, it made the CTest-Script depend on the Jenkins CI environment while it should be perfectly ok to just run the CTest-Script outside of a Jenkins env. Here, we are still within a single module/abstraction boundary. But your second argument is of course the relevant one, what, if we are too clever and hide it from ourselves :P

Reply:
So,
* move the condition to `tests/examples/simulation/Tutorial1/CMakeLists.txt`,
* and pass a flag `ENABLE_LD_LIBRARY_PATH_WORKAROUND` or similar?

I think I would want to avoid exposing the LD_LIBRARY_PATH content to the test suite CMakeLists, no?

Reply:
I like something like `ENABLE_LD_LIBRARY_PATH_WORKAROUND`. Long enough that we actually want to get rid of it once FairSoft / upstream has fixed all of this.

And still abstract enough for the caller.

Right, no explicit LD_LIBRARY_PATH handling there, unless we're absolutely forced to.